src/plugins/imagery/components/ImageryView.vue

```diff
@@ -670,23 +676,47 @@ export default {
                 this.$refs.thumbsWrapper.scrollLeft = scrollWidth;
             });
         },
+        matchIndexOfPreviousImage(previous, imageHistory) {
+            // match logic uses a composite of url and time to account
+            // for example imagery not having fully unique urls
+            return imageHistory.findIndex((x) => (
+                x.url === previous.url
+                && x.time === previous.time
+            ));
+        },
         setFocusedImage(index, thumbnailClick = false) {
+            let focusedIndex = index;
+            if (!(Number.isInteger(index) && index > -1)) {
+                return;
+            }
+
+            if (this.previousFocusedImage) {
+                // determine if the previous image exists in the new bounds of imageHistory
+                const matchIndex = this.matchIndexOfPreviousImage(
+                    this.previousFocusedImage,
+                    this.imageHistory
+                );
+                focusedIndex = matchIndex > -1 ? matchIndex : this.imageHistory.length - 1;
```

Comment:
If there is not a match on the previous image, it will set the focused image to the last item in the array. Not sure if that is the desired behavior.

Reply:
That's my understanding. Basically, it just reverts the most recent one, which makes sense.